web-registry/src/hooks/projects/useProjectWithMetadata.ts

```diff
@@ -0,0 +1,167 @@
+import { useCallback } from 'react';
+import { ProjectInfo } from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+
+import {
+  ProjectByIdQuery,
+  ProjectByOnChainIdQuery,
+  useUpdateProjectByIdMutation,
+} from 'generated/graphql';
+import { graphqlClient } from 'lib/clients/graphqlClient';
+import { getProjectKey } from 'lib/queries/react-query/ecocredit/getProjectQuery/getProjectQuery.constants';
+import { getMetadataQuery } from 'lib/queries/react-query/registry-server/getMetadataQuery/getMetadataQuery';
+import { getProjectByIdQuery } from 'lib/queries/react-query/registry-server/graphql/getProjectByIdQuery/getProjectByIdQuery';
+import { getProjectByIdKey } from 'lib/queries/react-query/registry-server/graphql/getProjectByIdQuery/getProjectByIdQuery.constants';
+import { getProjectByOnChainIdQuery } from 'lib/queries/react-query/registry-server/graphql/getProjectByOnChainIdQuery/getProjectByOnChainIdQuery';
+import { getProjectByOnChainIdKey } from 'lib/queries/react-query/registry-server/graphql/getProjectByOnChainIdQuery/getProjectByOnChainIdQuery.constants';
+
+import { ReturnType as ProjectEditSubmitReturnType } from 'pages/ProjectEdit/hooks/useProjectEditSubmit';
+import {
+  BasicInfoFormValues,
+  DescriptionValues,
+  ProjectLocationFormValues,
+} from 'components/organisms';
+import { MediaValues } from 'components/organisms/MediaForm';
+
+export type OffChainProject =
+  | ProjectByIdQuery['projectById']
+  | ProjectByOnChainIdQuery['projectByOnChainId'];
+
+interface Props {
+  projectId?: string;
+  isEdit?: boolean;
+  projectEditSubmit: ProjectEditSubmitReturnType;
+  navigateNext: () => void;
+  anchored?: boolean;
+  onChainProject?: ProjectInfo;
+}
+
+interface Res {
+  offChainProject?: OffChainProject;
+  metadata: any; // TODO update with proper type
```

Comment:
will do that once https://github.com/regen-network/regen-web/pull/1701 is merged to use new project metadata types, ditto for all related TODO comments